packages/eslint-plugin/src/rules/thenable-in-promise-aggregators.ts

```diff
@@ -0,0 +1,176 @@
+import {
+  isPromiseConstructorLike,
+  isPromiseLike,
+  isTypeAnyType,
+  isTypeUnknownType,
+} from '@typescript-eslint/type-utils';
+import type { TSESTree } from '@typescript-eslint/utils';
+import { AST_NODE_TYPES } from '@typescript-eslint/utils';
+import * as tsutils from 'ts-api-utils';
+import * as ts from 'typescript';
+
+import { createRule, getParserServices } from '../util';
+
+export default createRule({
+  name: 'thenable-in-promise-aggregators',
+  meta: {
+    docs: {
+      description:
+        'Disallow passing non-Thenable values to promise aggregators',
+      recommended: 'strict',
+      requiresTypeChecking: true,
+    },
+    messages: {
+      inArray:
+        'Unexpected non-Thenable value in array passed to promise aggregator.',
+      arrayArg:
+        'Unexpected array of non-Thenable values passed to promise aggregator.',
+    },
+    schema: [],
+    type: 'problem',
+  },
+  defaultOptions: [],
+
+  create(context) {
+    const services = getParserServices(context);
+    const checker = services.program.getTypeChecker();
+
+    const aggregateFunctionNames = ['all', 'race', 'allSettled', 'any'];
```

Comment:
[Style] Just nitpicking, but - what do you think about making this a `Set` and/or moving it outside of the rule's `create`? There's no need to remake it every time. And semantically, a `Set` is really more of what this is being used for (https://github.com/sindresorhus/eslint-plugin-unicorn/blob/bb9e7884c105f2c0f942153f46dcd3c4fe422482/docs/rules/prefer-set-has.md).

Reply:
Makes sense to me! For breaking it out it's also more consistent with `space-infix-ops` after some digging